Lib/test/test_zipfile.py

```diff
@@ -549,6 +549,14 @@ def test_add_file_before_1980(self):
         with zipfile.ZipFile(TESTFN2, "w") as zipfp:
             self.assertRaises(ValueError, zipfp.write, TESTFN)
 
+    def test_add_file_before_1980_no_strict_timestamps(self):
+        # Set atime and mtime to 1970-01-01
+        os.utime(TESTFN, (0, 0))
```

Comment:
Instead of using the real filesystem which can cause troubles, maybe you can mock os.stat() intsead of return the date that you want?

It may allow to test date after year 2107.

Reply:
The previous test is also using this. It shouldn't be a problem then.